models/src/controller.rs

```diff
@@ -212,18 +237,13 @@ pub fn controller_deployment(
                                 ..Default::default()
                             },
                             EnvVar {
-                                name: "MAX_CONCURRENT_UPDATE".to_string(),
-                                value: Some(max_concurrent_update),
+                                name: "SCHEDULER_CRON_EXPRESSION".to_string(),
+                                value: Some(scheduler),
                                 ..Default::default()
                             },
                             EnvVar {
-                                name: "UPDATE_WINDOW_START".to_string(),
```

Comment:
Won't this remove the `UPDATE_WINDOW_START` from our yaml definitions? If we're still expecting to consume these env vars for windows alongside cron expressions, we should keep these.

Reply:
No. The idea is we only allow one env that can be accepted by controller overall which is`SCHEDULER_CRON_EXPRESSION`, but that doesn't mean we can't consume `UPDATE_WINDOW_START ` and `UPDATE_WINDOW_STOP`. Because while building yaml, if users intend to use time window `UPDATE_WINDOW_START `, this [function](https://github.com/bottlerocket-os/bottlerocket-update-operator/pull/351/files#diff-66984611dec04515bc8bbfcf3e6877de3624052c36706ae22c87ac0e964432e2R317-R337) will automatically convert it to cron expression format. I was thinking it's better to keep only one env for scheduler and time window feature, because it can avoid users to specify two of them and also make us easy to remove all update time related codes in the future.

Reply:
> The idea is we only allow one env that can be accepted by controller overall which is `SCHEDULER_CRON_EXPRESSION`

Ahh, that makes sense.

> I was thinking it's better to keep only one env for scheduler and time window feature

Yes, I agree. If you need to do another update, might be worth calling that out in the docs abit more. Maybe something like

```diff
Note!! that these settings are deprecated and will be removed in a future release.
You should use the Scheduler settings below.
If you still decide to use these settings,
please use "hour:0:0" format only instead of "hour:minute:second".

+ If you use _both_ `SCHEDULER_CRON_EXPRESSION`
+ and the `UPDATE_*` env variables,
+ the controller will fail and exit early indication a mis-configuration
```

Reply:
I think the way this is written, the `UPDATE_WINDOW_*` settings can only be provided at build time, but most customers would be modifying their deployments to add the environment variables rather than re-generating the YAML, I expect.

Would it make sense for us to allow the controller to accept both and perform the conversion on startup, rather than doing so during generation of the YAML?

This may become more important when we merge the helm changes, I'm happy to followup with a fix.

Reply:
that makes sense! I'll upgrade it.